File: xkcd/src/main/java/xyz/jienan/xkcd/comics/contract/SingleComicContract.java
package xyz.jienan.xkcd.comics.contract;

import android.graphics.Bitmap;

import xyz.jienan.xkcd.base.BasePresenter;
import xyz.jienan.xkcd.base.BaseView;
import xyz.jienan.xkcd.model.XkcdPic;

public interface SingleComicContract {

    interface View extends BaseView<Presenter> {

        void explainLoaded(String result);

        void explainFailed();

        void renderXkcdPic(XkcdPic xkcdPicInDB);

        void setLoading(boolean isLoading);
    }

    interface Presenter extends BasePresenter {

        void loadXkcd(int index);

        void getExplain(long index);

        void updateXkcdSize(XkcdPic currentPic, Bitmap resource);
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/comics/contract/SingleComicContract.kt
package xyz.jienan.xkcd.comics.contract

import android.graphics.Bitmap

import xyz.jienan.xkcd.base.BasePresenter
import xyz.jienan.xkcd.base.BaseView
import xyz.jienan.xkcd.model.XkcdPic

interface SingleComicContract {

    interface View : BaseView<Presenter> {

        var translationMode : Int

        fun explainLoaded(result: String)

        fun explainFailed()

        fun renderXkcdPic(xkcdPic: XkcdPic)

        fun setLoading(isLoading: Boolean)

        fun setAltTextVisibility(gone: Boolean)
    }

    interface Presenter : BasePresenter {

        fun loadXkcd(index: Int)

        fun getExplain(index: Long)

        fun updateXkcdSize(xkcdPic: XkcdPic?, resource: Bitmap?)

        val showLocalXkcd : Boolean
    }
}
